custom_search_commands/python/customsearchcommands_app/README.md

```diff
@@ -0,0 +1,176 @@
+|:--------------|:-----------|:------------------------------------------------------------------------------------------------------|
+| countmatches  | Streaming  | Counts the number of non-overlapping matches to a regular expression in a set of fields.              |
+| generatetext  | Generating | Generates a specified number of events containing a specified text string.                            |
+| simulate      | Generating | Generates a sequence of events drawn from a csv file using repeated random sampling with replacement. |
+| generatehello | Generating | Generates a specified number of events containing the text string 'hello'.                            |
+| sum           | Reporting  | Adds all of the numbers in a set of fields.                                                           |
+| filter        | Eventing   | Filters records from the events stream based on user-specified criteria.                              |
+
+The app is tested on Splunk 8. Here is its manifest:
+
+```
+├── bin
+│   ├── countmatches.py .......... CountMatchesCommand implementation
+│   ├── generatetext.py .......... GenerateTextCommand implementation
+│   ├── simulate.py .............. SimulateCommand implementation
+│   └── sum.py ................... SumCommand implementation
+├── lib
+|   └── splunklib ................ splunklib module
+├── default
+│   ├── data
+│   │   └── ui
+│   │       └── nav
+│   │           └── default.xml ..
+│   ├── app.conf ................. Used by Splunk to maintain app state [1]
+│   ├── commands.conf ............ Search command configuration [2]
+│   ├── logging.conf ............. Python logging[3] configuration in ConfigParser[4] format
+│   └── searchbnf.conf ........... Search assistant configuration [5]
+└── metadata
+    └── default.meta ............. Permits the search assistant to use searchbnf.conf[6]
+```
+**References**  
+[1] [app.conf](https://docs.splunk.com/Documentation/Splunk/latest/Admin/Appconf) <br />
+[2] [commands.conf](https://docs.splunk.com/Documentation/Splunk/latest/Admin/Commandsconf) <br />
+[3] [Python Logging HOWTO](https://docs.python.org/2/howto/logging.html) <br />
+[4] [ConfigParser—Configuration file parser](https://docs.python.org/2/library/configparser.html) <br />
+[5] [searchbnf.conf](https://docs.splunk.com/Documentation/Splunk/latest/admin/Searchbnfconf) <br />
+[6] [Set permissions in the file system](https://docs.splunk.com/Documentation/Splunk/latest/AdvancedDev/SetPermissions#Set_permissions_in_the_filesystem) <br />
+
+## Installation
+
++ Bring up Dockerized Splunk with the app installed from the root of this repository via:
+
+### Step 1
+Execute the following command from the root of this repository.
+```shell
+SPLUNK_VERSION=latest docker compose up -d
+```
+
+### Step 2
+Check the container health, run:
+```shell
+docker ps
+```
+  
+### Step 3
```

Comment:
So there is already some precedent for this in the repo - see the `make modules` Makefile command: https://github.com/splunk/splunk-app-examples/blob/master/Makefile#L29

Though that will install splunklib into this directory: `vendor/splunklib`, actually the preferred guidance is to install into `lib/splunklib` which can be done using the same command but with `-t lib` instead of `-t vendor`.

Can we rework this guidance to be more similar to the following workflow by updating the Makefile to account for any examples that require the Python SDK to be installed to have it installed as part of `make modules`?
1. Run `make modules` (latest version of splunklib is downloaded from PyPi and copied to the `<app>/lib` directory.
2. Run `make up` to visit the example (example app(s) are already included as linked volumes as part of this file: https://github.com/splunk/splunk-app-examples/blob/master/docker-compose.yml#L15-L18)

This way fewer steps are needed for the user to take.

If there are too many apps to map at once to the volume we can consider splitting the docker-compose file into several.